cirq-core/cirq/value/condition_test.py

```diff
@@ -35,33 +35,35 @@ def test_key_condition_with_keys():
 
 def test_key_condition_str():
     assert str(init_key_condition) == '0:a'
+    assert str(cirq.KeyCondition(key_a, index=-2)) == '0:a[-2]'
 
 
 def test_key_condition_repr():
     cirq.testing.assert_equivalent_repr(init_key_condition)
+    cirq.testing.assert_equivalent_repr(cirq.KeyCondition(key_a, index=-2))
 
 
 def test_key_condition_resolve():
     def resolve(measurements):
         classical_data = cirq.ClassicalDataDictionaryStore(_measurements=measurements)
         return init_key_condition.resolve(classical_data)
 
-    assert resolve({'0:a': [1]})
-    assert resolve({'0:a': [2]})
-    assert resolve({'0:a': [0, 1]})
-    assert resolve({'0:a': [1, 0]})
-    assert not resolve({'0:a': [0]})
-    assert not resolve({'0:a': [0, 0]})
-    assert not resolve({'0:a': []})
-    assert not resolve({'0:a': [0], 'b': [1]})
+    assert resolve({'0:a': [[1]]})
```

Comment:
@95-martin-orion are you talking about this one (also linked in the current PR description): https://github.com/daxfohl/Cirq/compare/repeated...daxfohl:repeated2? I haven't made a PR of it yet since it depends on this one, and it needs work around serializing the new field in CircuitOp etc. But the core functionality is all in that diff and is surprisingly simple to implement. I'll flesh that out further and post it once this one is merged.

Reply:
Yup, that's the one I had in mind. I'll keep an eye out for the PR.